pkg/nmstate/nmstate.go

```diff
@@ -1 +1,223 @@
 package nmstate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+
+	nmstateV1 "github.com/nmstate/kubernetes-nmstate/api/v1"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Builder provides struct for the NMState object containing connection to
+// the cluster and the NMState definitions.
+type Builder struct {
+	Definition *nmstateV1.NMState
+	Object     *nmstateV1.NMState
+	apiClient  *clients.Settings
+	errorMsg   string
+}
+
+// NewBuilder creates a new instance of Builder.
+func NewBuilder(apiClient *clients.Settings, name string) *Builder {
```

Comment:
Do we have scenario when we need to create this object manually? I think we can remove this function.